prime.go

```diff
@@ -109,14 +109,14 @@ func (n *ProtoNode) MapIterator() ipld.MapIterator {
 // before itr.Done becomes true.
 func (n *ProtoNode) ListIterator() ipld.ListIterator {
 	_, _ = n.EncodeProtobuf(false)
-	return n.encoded.ListIterator()
+	return n.imNode.ListIterator()
 }
 
 // Length returns the length of a list, or the number of entries in a map,
 // or -1 if the node is not of list nor map kind.
 func (n *ProtoNode) Length() int64 {
 	_, _ = n.EncodeProtobuf(false)
-	return n.encoded.Length()
+	return n.imNode.Length()
```

Comment:
all these accesses on imNode need read locks, no?

Reply:
I opted to not lock anything in `prime.go`, hoping it will be used in a "smarter" context... Perhaps I should anyway ( this is *so much* locking... )

@Stebalien what do you think?